src/pages/index.js

```diff
@@ -22,96 +34,233 @@ const profileDescriptionInput = profileEditModal.querySelector(
 
-const renderCard = (cardData) => {
-  const card = createCard(cardData);
-  section.addItem(card);
-};
+// UserInfo
+const user = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+
+api
+  .getUserData()
+  .then((profileData) => {
+    if (profileData) {
+      user.setUserInfo(profileData.name, profileData.about);
+      user.changeAvatar(profileData.avatar);
+    }
+  })
+  .catch((err) => console.log("Error loading user info:", err));
 
 /* -------------------------------------------------------------------------- */
 /*                                  Popups                                    */
 /* -------------------------------------------------------------------------- */
-//Edit Profile Popup
+
+// Edit Profile Popup
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileEditSubmit
+  (profileData) => {
+    const submitButton = profileEditForm.querySelector(".modal__save");
+    const initialButtonText = submitButton.textContent;
+    submitButton.textContent = "Saving...";
+
+    api
+      .editProfile(profileData.title, profileData.subheader)
+      .then((updatedUserInfo) => {
+        user.setUserInfo(updatedUserInfo.name, updatedUserInfo.about);
+        editProfilePopup.close();
+      })
+      .catch((err) => console.log(err))
+      .finally(() => {
+        submitButton.textContent = initialButtonText;
+      });
+  }
 );
 editProfilePopup.setEventListeners();
 
-//Add New Card Popup
-const newCardPopup = new PopupWithForm("#add-card-modal", handleAddCardSubmit);
+// Add New Card Popup
+const newCardPopup = new PopupWithForm("#add-card-modal", (newCardData) => {
+  const submitButton = addCardForm.querySelector(".modal__save");
+  const initialButtonText = submitButton.textContent;
+  submitButton.textContent = "Saving...";
+
+  api
+    .addNewCard({ name: newCardData.title, link: newCardData.url })
+    .then((cardData) => {
+      renderCard(cardData);
+      newCardPopup.close();
+    })
+    .catch((err) => console.log(err))
+    .finally(() => {
+      submitButton.textContent = initialButtonText;
+    });
+});
 newCardPopup.setEventListeners();
 
-//Preview Image Popup
+// Preview Image Popup
 const previewImagePopup = new PopupWithImage("#preview-modal");
 previewImagePopup.setEventListeners();
 
-//Section
+// Section to render cards
 const section = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: renderCard,
   },
   ".cards__list"
 );
-section.renderItems();
 
-//UserInfo
-const user = new UserInfo(".profile__title", ".profile__description");
+api
+  .getInitialCards()
+  .then((cards) => {
+    section.renderItems(cards);
+  })
+  .catch((err) => console.log("Error fetching cards:", err));
+
+// Edit Avatar Popup
+const avatarEditPopup = new PopupWithForm(
+  "#profile-avatar-modal",
+  (formData) => {
+    const submitButton = avatarEditPopup._popupElement.querySelector(
```

Comment:
`._popupElement`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.



------------

By the way, you have `_formButton` there as the submit button. You can use it here instead of searching it again